nevergrad/functions/ml/mlfunctionlib.py

```diff
@@ -163,16 +151,10 @@ def __init__(
                 min_samples_split=p.Log(lower=0.0000001, upper=1),
                 regressor="decision_tree",
             )
-            # We use "partial" for fixing the parameters of the neural network, given that we work on the decision tree only.
-            super().__init__(partial(self._ml_parametrization, noise_free=False,
-                                     alpha=1.0, learning_rate="no", regressor="decision_tree",
-                                     activation="no", solver="no"), parametrization)
-            # For the test we just switch noise_free to True.
-            self.evaluation_function = partial(self._ml_parametrization, criterion="mse",  # type: ignore
-                                               min_samples_split=0.00001,
-                                               regressor="decision_tree", noise_free=not overfitter,
-                                               alpha=1.0, learning_rate="no",
-                                               activation="no", solver="no")
+            params = dict(noise_free=False,
+                          alpha=1.0, learning_rate="no", regressor="decision_tree",
+                          activation="no", solver="no")
+            evalparams = dict(params, criterion="mse", min_samples_split=0.00001)
```

Comment:
@teytaud this is probably not doing at all what you want:
You would like the evaluation to be mse only? But the parametrization may override these parameters, it's not the other way round, have you checked it or must it be checked?

Reply:
Ah thanks for pointing out this issue.
I've seen your other PR, I'll play with it, thanks for this as well.